bluesky_adaptive/tests/test_tsuchinoko_agents.py

```diff
@@ -0,0 +1,165 @@
+from tsuchinoko.execution.simple import SimpleEngine
+from tsuchinoko.execution.threaded_in_process import ThreadedInProcessEngine
+from tsuchinoko.utils.runengine import get_run_engine
+from typing import Tuple, Union
+from xarray import Dataset
+
+from bluesky_adaptive.agents.tsuchinoko import GPCamTsuchinokoAgent
+from bluesky_adaptive.utils.offline import OfflineAgent
+from ..typing import BlueskyRunLike
+
+
+# Disable logging to console when running tests
+# NOTE: it seems there is a bug between loguru and pytest; pytest tries to log to a tempfile, but closes it when finished
+# NOTE: if loguru has a backlog of messages
+# logger.remove()
+
+
+@fixture
+def image_data():
+    # Load data from scipy to be used as a luminosity map
+    image = datasets.face(gray=True)
+    return image
+
+
+@fixture
+def image_func(image_data):
+    # Bilinear sampling will be used to effectively smooth pixel edges in source data
+    def bilinear_sample(pos):
+        return pos, ndimage.map_coordinates(image_data, [[pos[1]], [pos[0]]], order=1)[0], 1, {}
+
+    return bilinear_sample
+
+
+@fixture
+def simple_execution_engine(image_func):
+    execution = SimpleEngine(measure_func=image_func)
+    return execution
+
+
+@fixture
+def gpcam_engine(image_data):
+    # Define a gpCAM adaptive engine with initial parameters
+    adaptive = GPCAMInProcessEngine(dimensionality=2,
+                                    parameter_bounds=[(0, image_data.shape[1]),
+                                                      (0, image_data.shape[0])],
+                                    hyperparameters=[255, 100, 100],
+                                    hyperparameter_bounds=[(0, 1e5),
+                                                           (0, 1e5),
+                                                           (0, 1e5)])
+    return adaptive
+
+
+@fixture
+def threaded_execution_engine(image_func):
+    execution = ThreadedInProcessEngine(image_func)
+    yield execution
+    execution.exiting = True
+    execution.measure_thread.join()
+
+@fixture
+def core(gpcam_engine, threaded_execution_engine):
+    adaptive_engine, execution_engine = gpcam_engine, threaded_execution_engine
+    logger.info('starting setup')
+    core = ZMQCore()
+    core.set_adaptive_engine(adaptive_engine)
+    core.set_execution_engine(execution_engine)
+    server_thread = Thread(target=core.main)
+    server_thread.start()
+    core.state = CoreState.Starting
+    logger.info('setup complete')
+
+    yield core
+
+    core.exit()
+    server_thread.join()
+    logger.info('teardown complete')
+
+
+class GPTestAgent(GPCamTsuchinokoAgent, OfflineAgent):
+    def measurement_plan(self, point: ArrayLike) -> Tuple[str, list, dict]:
+        return self.measurement_plan_name, [1.5], dict()
+
+    def unpack_run(self, run: BlueskyRunLike) -> Tuple[Union[float, ArrayLike], Union[float, ArrayLike]]:
+        self.counter += 1
+        return self.counter, np.random.rand(10)
```

Comment:
From above, the `np.random.rand` may need to return a (10,2) array, where the second dim is var.